evalml/model_understanding/metrics.py

```diff
@@ -282,27 +258,30 @@ def roc_curve(y_true, y_pred_proba):
                   * `auc_score`: The area under the ROC curve.
     """
     y_true_ww = infer_feature_types(y_true)
-    y_true_np = _convert_ww_series_to_np_array(y_true_ww)
-    y_pred_proba = infer_feature_types(y_pred_proba).to_numpy()
-
-    if len(y_pred_proba.shape) == 1:
-        y_pred_proba = y_pred_proba.reshape(-1, 1)
-    if y_pred_proba.shape[1] == 2:
-        y_pred_proba = y_pred_proba[:, 1].reshape(-1, 1)
-    nan_indices = np.logical_or(pd.isna(y_true_np), np.isnan(y_pred_proba).any(axis=1))
-    y_true_np = y_true_np[~nan_indices]
+    y_pred_proba = infer_feature_types(y_pred_proba)
+
+    # Standardize data to be a DataFrame even for binary target
+    if isinstance(y_pred_proba, pd.Series):
+        y_pred_proba = pd.DataFrame(y_pred_proba)
+    # Only use one column for binary inputs that are still a DataFrame
+    elif y_pred_proba.shape[1] == 2:
+        y_pred_proba = pd.DataFrame(y_pred_proba.iloc[:, 1])
```

Comment:
yeah, I also struggled to understand this, but it's specifically meant to allow users to pass in the results from binary classification `predict_proba` without having to pull out the positive class, since we don't make any such requirements for the multiclass case. It was suggested by freddy in this comment: https://github.com/alteryx/evalml/pull/1164#discussion_r488910475. We test it here: https://github.com/alteryx/evalml/blob/main/evalml/tests/model_understanding_tests/test_metrics.py#L347-L366

I tried to add some clarity with the comment `Only use one column for binary inputs that are still a DataFrame` but let me know if there's a better way to clarify what's going on here